mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresMessageModule.java

```diff
@@ -0,0 +1,122 @@
+                .constraint(DSL.primaryKey(MESSAGE_ID))
+                .comment("Holds the metadata of a mail")))
+            .supportsRowLevelSecurity();
+    }
+
+    interface MessageToMailboxTable {
+        Table<Record> TABLE_NAME = DSL.table("message_mailbox");
+        Field<UUID> MAILBOX_ID = DSL.field("mailbox_id", SQLDataType.UUID.notNull());
+        Field<Long> MESSAGE_UID = DSL.field("message_uid", SQLDataType.BIGINT.notNull());
+        Field<UUID> MESSAGE_ID = PostgresMessageModule.MESSAGE_ID;
+        Field<String> THREAD_ID = DSL.field("thread_id", SQLDataType.NVARCHAR(255));
+        Field<Boolean> IS_DELETED = DSL.field("is_deleted", SQLDataType.BOOLEAN.nullable(false)
+            .defaultValue(DSL.field("false", SQLDataType.BOOLEAN)));
+        Field<Boolean> IS_ANSWERED = DSL.field("is_answered", SQLDataType.BOOLEAN.nullable(false));
+        Field<Boolean> IS_DRAFT = DSL.field("is_draft", SQLDataType.BOOLEAN.nullable(false));
+        Field<Boolean> IS_FLAGGED = DSL.field("is_flagged", SQLDataType.BOOLEAN.nullable(false));
+        Field<Boolean> IS_RECENT = DSL.field("is_recent", SQLDataType.BOOLEAN.nullable(false));
+        Field<Boolean> IS_SEEN = DSL.field("is_seen", SQLDataType.BOOLEAN.nullable(false));
+        Field<Boolean> IS_USER = DSL.field("is_user", SQLDataType.BOOLEAN.nullable(false));
+        Field<String[]> USER_FLAGS = DSL.field("user_flags", DataTypes.STRING_ARRAY);
+
+        PostgresTable TABLE = PostgresTable.name(TABLE_NAME.getName())
+            .createTableStep(((dsl, tableName) -> dsl.createTableIfNotExists(tableName)
+                .column(MAILBOX_ID)
+                .column(MESSAGE_UID)
+                .column(MESSAGE_ID)
+                .column(THREAD_ID)
+                .column(IS_DELETED)
+                .column(IS_ANSWERED)
+                .column(IS_DRAFT)
+                .column(IS_FLAGGED)
+                .column(IS_RECENT)
+                .column(IS_SEEN)
+                .column(IS_USER)
+                .column(USER_FLAGS)
+                .constraint(DSL.primaryKey(MAILBOX_ID, MESSAGE_UID))
+                .comment("Holds mailbox and flags for each message")))
+            .supportsRowLevelSecurity();
+
+        PostgresIndex MESSAGE_ID_INDEX = PostgresIndex.name("message_mailbox_message_id_index")
+            .createIndexStep((dsl, indexName) -> dsl.createIndexIfNotExists(indexName)
+                .on(TABLE_NAME, MESSAGE_ID));
+
+        PostgresIndex MAILBOX_ID_MESSAGE_UID_INDEX = PostgresIndex.name("mailbox_id_mail_uid_index")
+            .createIndexStep((dsl, indexName) -> dsl.createIndexIfNotExists(indexName)
+                .on(TABLE_NAME, MAILBOX_ID, MESSAGE_UID.asc()));
+    }
+
+    PostgresModule MODULE = PostgresModule.builder()
+        .addTable(MessageTable.TABLE)
+        .addTable(MessageToMailboxTable.TABLE)
+        .addIndex(MessageToMailboxTable.MESSAGE_ID_INDEX)
+        .addIndex(MessageToMailboxTable.MAILBOX_ID_MESSAGE_UID_INDEX)
```

Comment:
We will likely need more index on this...

Out of my head...

 - mailbox_id + IS_SEEN(ASC)
 - mailbox_id + IS_RECENT
 - mailbox_id + IS_DELETED

Reply:
Where we have Cassandra denormalisation table we NEED to have indexes. I know this for sure. Performance study already done.